### PR TITLE
[Verification] Adds functional test for sealing and verification

### PR DESCRIPTION
**Pull request by yhassanzadeh13**
This PR adds a functional test for the entire pipeline of sealing and verification based on result approvals. It deploys a contract on `testnet` hence causing execution result for more than a single (system chunk), and then evaluates whether the verification node who has been assigned all chunks of that result can generate result approvals for each of those chunks. Most importantly, it evaluates whether consensus nodes can seal the corresponding block of that execution result based on the result approvals of the verification node.

### CODE REVIEW

File: integration/Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ integration-test: common-tests execution-tests verification-tests collection-tes
 
 # NOTE: Currently skipping collection and consensus tests due to them not passing properly on CI
 .PHONY: ci-integration-test
-ci-integration-test: common-tests execution-tests # collection-tests # consensus-tests # verification-tests
+ci-integration-test: common-tests execution-tests verification-tests # collection-tests # consensus-tests
```
Comment (yhassanzadeh13) [inline integration/Makefile:15]: Enables verification integration tests.

File: integration/tests/verification/suite.go [hunk 2]
```diff
@@ -25,8 +25,9 @@ type Suite struct {
 	nodeConfigs                []testnet.NodeConfig // used to keep configuration of nodes in testnet
 	nodeIDs                    []flow.Identifier    // used to keep identifier of nodes in testnet
 	ghostID                    flow.Identifier      // represents id of ghost node
-	exeID                      flow.Identifier      // represents id of execution node
-	verID                      flow.Identifier      // represents id of verification node
+	exe1ID                     flow.Identifier
+	exe2ID                     flow.Identifier
```
Comment (yhassanzadeh13) [inline integration/tests/verification/suite.go:29]: Even on sealing based on result approvals, we need more than one agreeing-on execution result for a block to be sealed. Hence, we need more than one execution node.

File: integration/tests/verification/suite.go [hunk 3]
```diff
@@ -75,6 +76,8 @@ func (s *Suite) SetupTest() {
 			testnet.WithID(nodeID),
 			testnet.WithLogLevel(zerolog.FatalLevel),
 			testnet.WithAdditionalFlag("--hotstuff-timeout=12s"),
+			testnet.WithAdditionalFlag("--required-verification-seal-approvals=1"),
+			testnet.WithAdditionalFlag("--required-construction-seal-approvals=1"),
```
Comment (yhassanzadeh13) [inline integration/tests/verification/suite.go:80]: We require at least one result approval for each chunk to be sealed.

File: integration/tests/verification/sealing_happypath_test.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+func TestSealingHappyPathTest(t *testing.T) {
+	suite.Run(t, new(TestSealingHappyPathTestSuite))
+}
+
+type TestSealingHappyPathTestSuite struct {
+	Suite
+}
+
+// TestSealingAndVerificationHappyPath evaluates the health of the happy path of verification and sealing. It
+//deploys a transaction into the testnet hence causing an execution result with more than
```
Comment (smnzhu) [inline integration/tests/verification/sealing_happypath_test.go:23]:
```suggestion
// deploys a transaction into the testnet hence causing an execution result with more than
```

File: integration/tests/verification/sealing_happypath_test.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	receiptB1 := s.ReceiptState.WaitForReceiptFrom(s.T(), blockB.Header.ID(), s.exe1ID)
+	s.T().Logf("receipt for blockB generated by execution node-1: %x result ID: %x", s.exe1ID, receiptB1.ExecutionResult.ID())
+	receiptB2 := s.ReceiptState.WaitForReceiptFrom(s.T(), blockB.Header.ID(), s.exe2ID)
+	s.T().Logf("receipt for blockB generated by execution node-2: %x result ID: %x", s.exe2ID, receiptB2.ExecutionResult.ID())
+
+	require.Equal(s.T(), receiptB1.ExecutionResult.ID(), receiptB2.ExecutionResult.ID(), "execution fork happened at blockB")
+	resultB := receiptB1.ExecutionResult
+	resultBId := resultB.ID()
+	// re-evaluates that resultB has more than one chunk.
+	require.Greater(s.T(), len(resultB.Chunks), 1)
```
Comment (smnzhu) [inline integration/tests/verification/sealing_happypath_test.go:52]: Just curious why this step is necessary?

**Review by jordanschalm — APPROVED** [review]

File: integration/tests/verification/sealing_happypath_test.go (hunk 5 above)
Comment (yhassanzadeh13) [reply integration/tests/verification/sealing_happypath_test.go:52]: We are running this test under the assumption that if a block changes state it should have more than a single chunk (i.e., at least one chunk + one system chunk). It is safer to examine this assumption before moving forward with test evaluation. This test is meant for the verification of results with more than one chunk. Agree that is not necessary, but safer.

**Review by zhangchiqing — APPROVED** [review]
Looks good

**Pull request merged**